src/js/actions/layers.js

```diff
@@ -1377,45 +1382,147 @@ define(function (require, exports) {
     };
 
     /**
+     * Helper function to find the right most artboard 
+     * @param {Immutable.List.of<Layer>} artboards
+     * @return {Layer}
+     */
+
+    var _getRightMost = function (artboards) {
+        var layer = artboards.reduce(function (selectedLayer, currentLayer) {
+                if (currentLayer.bounds.right > selectedLayer.bounds.right) {
+                    return currentLayer;
+                } else {
+                    return selectedLayer;
+                }
+            }, artboards.first());
+        return layer;
+    };
+
+    /**
+     * Helper function to get Bounds from specs (template preset)
+     * @param {string} specs, {Immutable.List.of<Layer>} artboards, {Layer} layerRef
+     * @return {Bounds} finalBounds
```

Comment:
If it is a uniform array, yes. If it's not a uniform array, then you shouldn't use an array; you should use an object instead, and declare it _exactly_ like this:
```
@param {{foo: number, bar: string, baz: object}}`
```
(Yes, there are two sets of curly braces. See [here](https://developers.google.com/closure/compiler/docs/js-for-compiler?hl=en#types) for the explanation.)